### DISCUSSION TITLE
有关peer选择的问题

### DISCUSSION THREAD

**Discussion by novada0** [SUSTech-CS305-Fall22/CS305-Project-Skeleton#39] [Q&A]
考虑这样的情况：peerA向peerB发送WHOHAS chunkhash1,chunkhash2，peerB返回IHAVE chunckdata1,chunkdata2。那么peerA可以向peerB发送GET chunkhash1，chunkhash2吗？还是说peerA只能向一个peer请求一个chunk?

**Comment by TANSixu** [comment]
Hi，一次get只能请求一个chunk。chunk1和chunk2都可以向peerB下载，但是需要两次get噢～
关于握手报文的讨论还可以参考 #24

**Reply by novada0** [reply]
好的好的，谢谢学长！那这种情况也要求concurrent下载吗？

**Reply by TANSixu** [reply]
不需要的哈，concurrency针对的是不同的peer噢～ 在上面的例子里，peerA只能向peerB下载完1后再下载2

**Reply by novada0** [reply]
这种情况的话，如果把chunkhash放在header，然后再根据from_address是不是就可以实现concurrency呢？

**Reply by TANSixu** [reply]
> 这种情况的话，如果把chunkhash放在header，然后再根据from_address是不是就可以实现concurrency呢？

是的，可以这样实现！需要注意的是，header可以拓展（在已有部分后面加新的信息），但是已有部分是不能改的噢～
